### PR TITLE
Feature/tekton operator

### LINKED ISSUE #49: Setup tekton
Setup terraform module for tekton.

The following will be required:
- [ ] Create module for `tekton pipelines` and `tekton triggers` (don't use `tekton-operator`, there seems to be issues when uninstalling/installing it)
- [ ] Create PR for tekton to add securityContext and volume mount to `initContainers` (or a mutating webhook using `opa`/`opa-gatekeeper`)

Putting issue on backlog.

### PR DESCRIPTION
**Pull request by simongottschlag**
fixes #49

### CODE REVIEW

File: modules/azure/aks/modules.tf [hunk 1]
```diff
@@ -213,3 +213,13 @@ module "velero" {
   azure_client_id                 = var.velero.identity.client_id
   azure_resource_id               = var.velero.identity.resource_id
 }
+
+# Velero
+module "tekton-operator" {
```
Comment (NissesSenap) [inline modules/azure/aks/modules.tf:218]: Does this have anything to do with velero?

File: validation/kubernetes/velero/main.tf [hunk 2]
```diff
@@ -1,12 +1,20 @@
 terraform {
```
Comment (NissesSenap) [inline validation/kubernetes/velero/main.tf:1]: Should this file be changed in this PR

**Review by NissesSenap — COMMENTED** [review]
N00b questions :)

File: modules/azure/aks/modules.tf (hunk 1 above)
Comment (simongottschlag) [reply modules/azure/aks/modules.tf:218]: Fixed!

File: validation/kubernetes/velero/main.tf (hunk 2 above)
Comment (simongottschlag) [reply validation/kubernetes/velero/main.tf:1]: Not really, but fixed it since I saw that I've missed it. ;)

**Comment by NissesSenap** [comment]
I know this is not what you are asking but in your task definition don't use: "inputs".
```
  steps:
  - name: product
    image: bash:latest
    script: |
      #!/usr/bin/env bash
      echo -n $(( "$(inputs.params.a)" * "$(inputs.params.b)" ))
```
It's not needed any more.
Just $(params.b) is enough

You also don't need the " around $(params.b) in your echo. And you have already defined that it's a string that you send in with your type.

**Comment by simongottschlag** [comment]
Will abandon this PR and re-open at a later stage. We need to get tekton `initContainers` to work together with opa-gatekeeper policies and move away from tekton-operator to setup tekton pipelines and tekton triggers directly with terraform.

**Pull request closed without merging**